esrally/resources/track-schema.json

```diff
@@ -207,6 +207,61 @@
       "type": "object",
       "description": "Meta-information which will be added to each metrics-record of this track."
     },
+    "component_templates": {
+      "type": "array",
+      "minItems": 1,
+      "uniqueItems": true,
+      "items": {
+        "title": "Component Template",
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "Component template name."
+          },
+          "template": {
+            "type": "string",
+            "description": "Component template file name."
+          }
+        },
+        "required": [
+          "name",
+          "template"
+        ]
+      }
+    },
+    "composable_templates": {
```

Comment:
By convention we use hyphens for user-facing names if possible (sometimes it's not). Can we rename this to `composable-templates`? Also, the loader code actually uses the hyphenated name already.